contrib/kl_layer/kl_layer.py

```diff
@@ -0,0 +1,189 @@
+"""
+    Usage example for a custom KL loss layer
+    Takes as an input two arrays: z_mean and z_log_var
+    and computes KL "distance" between normal distribution
+    and Gaussian with mu=z_mean and sigma=z_log_var
+
+    The HLS part is in hls4ml/templates/vivado/nnet_utils/nnet_distance.h
+"""
+from pathlib import Path
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.keras.layers.merge import _Merge as Merge
+from tensorflow.python.keras.utils import tf_utils
+from tensorflow.python.ops import math_ops
+
+import hls4ml
+from hls4ml.converters.keras_to_hls import parse_default_keras_layer
+from hls4ml.model.types import FixedPrecisionType, NamedType
+
+test_root_path = Path(__file__).parent
+
+
+# Keras implementation of a KL layer
+class Distance(Merge):
+    def _check_inputs(self, inputs):
+        if len(inputs) not in [2, 3]:
```

Comment:
In the end we never used the other losses, so there's no need for `Distance` class itself, nor the check for the number of inputs.

Reply:
ok, `Distance` class removed